packages/flutter/lib/src/cupertino/dialog.dart

```diff
@@ -196,110 +208,265 @@ class CupertinoAlertDialog extends StatelessWidget {
       children.add(const Padding(padding: const EdgeInsets.only(top: 8.0)));
     }
 
+    return new Container(
+      key: const Key('cupertino_alert_dialog_content_section'),
+      color: _kDialogColor,
+      child: new Column(
+        mainAxisSize: MainAxisSize.min,
+        crossAxisAlignment: CrossAxisAlignment.stretch,
+        children: children,
+      ),
+    );
+  }
+
+  Widget _buildActions() {
+    Widget actionSection = new Container(
+      height: 0.0,
+    );
     if (actions.isNotEmpty) {
-      final Widget actionSection = new _CupertinoAlertActionSection(
+      actionSection = new _CupertinoAlertActionSection(
         children: actions,
         scrollController: actionScrollController,
       );
-      children.add(
-        new Flexible(child: actionSection),
-      );
     }
 
-    return new Padding(
-      padding: const EdgeInsets.symmetric(vertical: _kEdgePadding),
-      child: new CupertinoDialog(
-        child: new Column(
-          mainAxisSize: MainAxisSize.min,
-          crossAxisAlignment: CrossAxisAlignment.stretch,
-          children: children,
+    return actionSection;
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return new Center(
+      child: new Container(
+        margin: const EdgeInsets.symmetric(vertical: _kEdgePadding),
+        width: _kCupertinoDialogWidth,
+        // The following clip is critical. The BackdropFilter needs to have
+        // rounded corners, but SKIA cannot internally create a rounded rect
+        // shape. Therefore, we have no choice but to clip, ourselves.
```

Comment:
it can't create a rounded-rect blur. It can create rounded-rect shapes. :-)

Reply:
corrected